parkings/api/operator/parking.py

```diff
@@ -30,12 +30,22 @@ class Meta:
             'location': {'default': None},
             'terminal_number': {'default': ''},
             'time_end': {'default': None},
+            'is_disc_parking': {'default': False},
         }
 
     def __init__(self, *args, **kwargs):
         super(OperatorAPIParkingSerializer, self).__init__(*args, **kwargs)
         self.fields['time_start'].timezone = pytz.utc
         self.fields['time_end'].timezone = pytz.utc
+        self.fields['zone'].required = True
+
+        self._set_required_extra_fields()
+
+    def _set_required_extra_fields(self):
+        # if it is disc parking, make location required and zone optional field.
+        if getattr(self, 'initial_data', None) and self.initial_data.get('is_disc_parking', False):
+            self.fields['location'].required = True
+            self.fields['zone'].required = False
```

Comment:
Use a separate variable for the getattr result to make this code cleaner:
```python
initial_data = getattr(self, 'initial_data', None)
if initial_data and initial_data.get('is_disc_parking'):
```

Also, remove the comment, as it's saying just the same thing as can be read from the code.